### PR TITLE
Support 64bit Way Ids Again

### PR DESCRIPTION
**Pull request by kevinkreiser**
osm wayids are technically 64bits. while in practice wayids dont usually appear above 32bits (yet) its entirely within spec to support this. we have encountered osm extracts for example who have ids above 32bits in size. this PR works by optionally adding 4 more bytes to the edgeinfo struct when it encounters a wayid that is larger than 32bits. for the osm planet this should mean that we see no increase in data size but only on extracts that actually make use of ids wider than 32bits

### CODE REVIEW

File: valhalla/baldr/edgeinfo.h [hunk 1]
```diff
@@ -208,6 +209,9 @@ class EdgeInfo {
   // The encoded shape of the edge
   const char* encoded_shape_;
 
+  // Where we optionally keep the other half of a 64bit wayid
+  const uint32_t* extended_wayid_;
```
Comment (kevinkreiser) [inline valhalla/baldr/edgeinfo.h:213]: we could make this even cheaper by using the spare in `w0_` and `item_`. if we did that we would only have to have a 16bit value here.

Comment (kevinkreiser) [reply valhalla/baldr/edgeinfo.h:213]: the downside would be slightly more complex logic in the setter and getter

**Comment by dnesbitt61** [comment]
could 64 bit way Ids be stored in a separate structure so that those who want smaller data (and work with 32 bit way Ids) can utilize them? The upper 32 bits for instance could be stored as a byte array after the shape string. This allows 4-byte words in EdgeInfo keeping the base structure smaller.

**Comment by kevinkreiser** [comment]
this is deprecated by #2422

**Pull request closed without merging**